### ISSUE TITLE
detect and resolve CGlib proxied controllers

### ISSUE THREAD

**Issue by binlecode** [binlecode/swagger#16]
did a deep dive into spring Aop via CGLIB on AoP annotations conflicting with Swagger annotations:
- Swagger ImplicitParams resolution is detecting native class methods against its ApiOperation annotation (GET, POST, etc), so that swagger UI ajax can link the params set to each Http method.
- the problem comes from the AoP linked annotation being leveraged by Aop, so that the host class (in this case the controller class) is automatically proxied by spring CGLib (as Grails controller is not interface based, but traits based), the ApiOperation GET/POST to params set match is broken
as swagger is only working on Grails controllers, the ugly easy get-around is to avoid logging execution time on action directly, but on service methods (which is somewhat preferred, as action timing tracking can be done decently in grails interceptors, you can have per controller interceptor by naming matching convention, or by controller groups, the mapping config is trivial).
- a side explanation of why other Grails annotations on controller doesn’t kill Swagger annotaion: that’s because they are not coupled with AoP, since those stock annotations are not proxied, Spring is not using CGLib, but JDK native bean replacing, instead of proxy-inheritance.

In summary, the CGLib proxy mutated the controller singleton bean signature, which leads to the mismatch between Swagger ApiOperation annotation and ImplicitParams annotation. 

Therefore, Swagger doc generation should detect CGlib proxied controller and resolve the target class behind the proxy to support the matching between ApiOperation and ImplicitParams annotation.

**Comment by binlecode** [comment]
included in version 1.2.4
